alerting/src/main/kotlin/org/opensearch/alerting/util/AlertingUtils.kt

```diff
@@ -163,6 +166,16 @@ inline fun <T : ThreadContext.StoredContext, R> T.use(block: (T) -> R): R {
     }
 }
 
+fun getCancelAfterTimeInterval(): Long {
+    // The default value for the cancelAfterTimeInterval is -1 and so, in this case
+    // we should ignore processing on the value
+    val givenInterval = MonitorRunnerService.monitorCtx.cancelAfterTimeInterval!!.minutes
+    if (givenInterval == -1L) {
```

Comment:
Does -1 get interpreted as no cancelAfterTimeout? I'm wondering if we should have a sensible default instead of defaulting to no cancelAfter. Could you elaborate on what conditions would necessitate a user changing this setting from -1 to a specific value?

Reply:
yes -1 is a default for [cancel_after_time_interval](https://opensearch.org/docs/latest/api-reference/search/#url-parameters).  Suppose `search.cancel_after_time_interval ` is set at cluster with some higher value and the plugin overrides it to a lower value. Thus, max(cluster setting search.cancel_after_time_interval, plugin's default minimum value) should be used while overriding before calling _search from plugin.